### PR TITLE
Two track surveys

### LINKED ISSUE #46: Implement "two-track" survey creation
Intercept and observational surveys have different requirements. These differences emerge, particularly, in how the surveys handle "counted" data, e.g., "how many people are riding bikes in the park?" vs. "did you ride a bike in the park?" 

Currently, the survey builder offers both types of questions (single vs. multiple count). 

Screen Shot 2019-03-11 at 2 29 32 PM

This can be confusing or "clunky." Instead, the form elements should depend on the type of survey. 

**On the `create` view, let's give users the option to chose one of two "tracks": observational or intercept.**

****

**Important note!** The PLDP handles observational data, but not intercept. For now, users who follow the observational track will also adhere to a strict implementation of PLDP. Alternatively, the intercept surveys will have greater flexible, since intercept surveys aim to capture fine-grained detail, gather opinions, and record non-observable facts.

### PR DESCRIPTION
**Pull request by beamalsky**
## Overview

This is the last bit of work needed to close #46! https://github.com/datamade/django-fobi/pull/2 will need to be merged before it passes checks. 

### To do 

- [x] Write a test or two!

## Testing Instructions

 * If you don't already have it, create the database: `pg_restore -C -j4 --no-owner just-spaces.dump | psql`
* Migrate: `python manage.py migrate`
* Login as testuser (password in the DataMade LastPass under "Just Spaces Staging")
* Create a new survey with type Observational
* When you go to add a new question, do you see the "Observational" category?
* Create a new survey with type Intercept
* When you go to add a new question, do you NOT see the "Observational" category?

Connects #46

### CODE REVIEW

File: justspaces/settings.py [hunk 1]
```diff
@@ -151,4 +151,4 @@
 
 FOBI_DEFAULT_THEME = 'foundation5'
 FOBI_THEME_FOOTER_TEXT = gettext.gettext('')
-FOBI_RESTRICT_PLUGIN_ACCESS = False
+FOBI_RESTRICT_PLUGIN_ACCESS = True
```
Comment (jeancochrane) [inline justspaces/settings.py]: I'm not totally confident I get what this is doing -- from the docs:

> FOBI_RESTRICT_PLUGIN_ACCESS (bool): If set to True, (Django) permission system for dash plugins is enabled. Defaults to True. Setting this to False makes all plugins available for all users.

Does this mean that some users won't have access to some plugins, or that plugins can only be used on the group of which they're a part?

Comment (beamalsky) [reply justspaces/settings.py]: I changed this when I was trying to figure out how to approach this issue and forgot to change it back—my bad! I think we shouldn't use Fobi's permissions system but should instead build our own once we get there.

**Comment by beamalsky** [comment]
@jeancochrane I've added a column in the list view to indicate if a survey is observational or intercept! Now that I've merged these [changes](https://github.com/datamade/django-fobi/pull/2) in `django-fobi`, you should see the right filtered categories in the form element dropdown. Could you give it another shot?

**Comment by beamalsky** [comment]
For these new tests to work, I made the user fixture a superuser. We'll probably want to change that once we address #48

File: tests/test_views.py [hunk 2]
```diff
@@ -26,6 +26,33 @@ def test_survey_list_run(client, user, survey_form_entry):
...
+@pytest.mark.django_db
+def test_survey_edit_intercept(client, user, survey_form_entry):
+    client.force_login(user)
+    url = reverse('fobi.edit_form_entry', kwargs={'form_entry_id': survey_form_entry.id})
+    response = client.get(url)
+
+    plugins = response.context['user_form_element_plugins']
+
+    print(plugins)
+    print(survey_form_entry.type)
```
Comment (jeancochrane) [inline tests/test_views.py]: Reminder to delete these print statements before bringing this in.

File: tests/test_views.py [hunk 3]
```diff
@@ -26,6 +26,33 @@ def test_survey_list_run(client, user, survey_form_entry):
...
+@pytest.mark.django_db
+def test_survey_edit_observational(client, user, survey_form_entry_observational):
+    client.force_login(user)
+    url = reverse('fobi.edit_form_entry', kwargs={'form_entry_id': survey_form_entry_observational.id})
+    response = client.get(url)
+
+    plugins = response.context['user_form_element_plugins']
+
+    assert response.status_code == 200
+    assert len(plugins) == 4
```
Comment (jeancochrane) [inline tests/test_views.py]: Nice tests!

**Pull request merged**